flake.nix

```diff
@@ -1,55 +1,72 @@
 {
   description = "flora";
   inputs = {
-    flake-utils = { url = "github:numtide/flake-utils"; };
-    horizon-platform = {
-      url = "git+https://gitlab.horizon-haskell.net/package-sets/horizon-platform";
-    };
+    flake-utils.url = "github:numtide/flake-utils";
+    horizon-platform.url =
+      "git+https://gitlab.horizon-haskell.net/package-sets/horizon-platform";
     nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
-    streamly.url = "git+https://github.com/composewell/streamly";
-    streamly.flake = false;
-    poolboy.url = "git+https://github.com/blackheaven/poolboy";
+    nixpkgs-souffle_2_2.url = "github:nixos/nixpkgs/nixos-22.05";
+    streamly = {
+      url = "github:composewell/streamly";
+      flake = false;
+    };
+    poolboy.url = "github:blackheaven/poolboy";
   };
-  outputs =
-    inputs@{ self, flake-utils, horizon-platform, nixpkgs, ... }:
+  outputs = inputs@{ self, flake-utils, horizon-platform, nixpkgs, ... }:
     flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
-    let
-      pkgs = import nixpkgs { inherit system; };
-      hsPkgs = with pkgs.haskell.lib;
-        horizon-platform.legacyPackages.${system}.override {
-          overrides = hfinal: hprev: rec {
-            flora = overrideCabal
-              (dontHaddock (dontCheck
-                (doJailbreak (hfinal.callCabal2nix "flora" ./. { }))))
-              (drv: {
-                preConfigure = ''
-                  cd cbits; ${pkgs.souffle}/bin/souffle -g categorise.{cpp,dl}
-                  cd ..
-                '';
-              });
-            streamly-core = hfinal.callCabal2nix "streamly-core" "${inputs.streamly}/core/" { };
-            streamly = hfinal.callCabal2nix "streamly" inputs.streamly { };
-            poolboy = hfinal.callCabal2nix "poolboy" inputs.poolboy { };
+      let
+        pkgs = import nixpkgs { inherit system; };
+        souffle_2_2 =
+          inputs.nixpkgs-souffle_2_2.legacyPackages.${system}.souffle;
+        hsPkgs = with pkgs.haskell.lib;
+          horizon-platform.legacyPackages.${system}.override {
+            overrides = hfinal: hprev: {
+              flora = overrideCabal (dontHaddock (dontCheck
```

Comment:
I know the change doesn't come from you but I wanted to note that it's weird that we have to jailbreak flora itself, at least flora itself should get proper dependencies if possible.

Reply:
Yeah, you're right. Especially since flora only has 2 or 3 actual version constraints in its cabal file. We might want to revisit this at a later point, for instance when trying to finally get it to `nix build`. (But that might not be worth the trouble: if the freeze file dictates the versions to be used, then any "normal" `haskellPackages` set will have problems fitting to that.)